src/vuestic-theme/vuestic-components/va-rating/VaStar.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+  <div class="va-star">
+    <i
+      :class="iconClassesComputed"
+      :style="iconStyles"
+      class="va-star__icon"
+      @mousemove="onHover"
+      @mouseleave="removeHover"
+      @click="rate()"
+      width="50"
+    />
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'va-star',
+  props: {
+    value: {
+      type: Boolean
+    },
+    icon: {
+      type: String,
+      default: 'fa fa-star'
+    },
+    isRatingHover: {
+      type: Boolean
+    },
+    halfIcon: {
+      type: String
+    },
+    isHalf: {
+      type: Boolean
+    },
+    iconClasses: {
+      type: String
+    },
+    iconStyles: {
+      type: Object
+    },
+    color: {
+      type: String
+    },
+    emptyIcon: {
+      type: String
+    },
+    hover: {
+      type: Boolean
+    }
+  },
+  data () {
+    return {
+      extraClasses: '',
+      hoverValue: ''
+    }
+  },
+  computed: {
+    iconClassesComputed () {
+      if (this.halfIcon) {
```

Comment:
early return + probably split into methods/helpers.
This is not human readable.